web/yo/app/scripts/controllers/gene.js

```diff
@@ -3302,6 +3435,55 @@ angular.module('oncokbApp')
         }]
     )
 
+    .controller('ReleaseGeneDialogCtrl', function($scope, $modalInstance, data) {
+        $scope.statusChecks = data.statusChecks || [];
+        $scope.error='';
+
+        _.each($scope.statusChecks, function(statusCheck) {
+            if (!statusCheck.status) {
```

Comment:
The logic here is
- if the statusCheck item already has a status which means the logic is decided outside of the controller. Nothing really needs to be done here.
- If the promise is not given, it should be an optional check. So assign a warning status.
- Depends on the promise status to assign the statusCheck status